packages/cmf/src/sagaRouter/matchPath.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Beware! Do not modify. Forked from react-router V4
+ * Will be available as a dependency
+ */
+
+import pathToRegexp from 'path-to-regexp';
+
+const patternCache = {};
+const cacheLimit = 10000;
+let cacheCount = 0;
+
+const compilePath = (pattern, options) => {
+	const cacheKey = `${options.end}${options.strict}`;
+	const cache = patternCache[cacheKey] || (patternCache[cacheKey] = {});
+
+	if (cache[pattern]) return cache[pattern];
+
+	const keys = [];
+	const re = pathToRegexp(pattern, keys, options);
+	const compiledPattern = { re, keys };
+
+	if (cacheCount < cacheLimit) {
+		cache[pattern] = compiledPattern;
+		cacheCount += 1;
+	}
+
+	return compiledPattern;
+};
+
+/**
+ * Public API for matching a URL pathname to a path pattern.
+ */
+const matchPath = (pathname, options = {}) => {
+	if (typeof options === 'string') options = { path: options }; // eslint-disable-line no-param-reassign
+
+	const { path = '/', exact = false, strict = false } = options;
+	const { re, keys } = compilePath(path, { end: exact, strict });
+	const match = re.exec(pathname);
+
+	if (!match) return null;
+
+	const [url, ...values] = match;
+	const isExact = pathname === url;
+
+	if (exact && !isExact) return null;
+
+	return {
+		path, // the path pattern used to match
+		url: path === '/' && url === '' ? '/' : url, // the matched portion of the URL
+		isExact, // whether or not we matched exactly
+		params: keys.reduce((memo, key, index) => {
```

Comment:
you can use this:

params: keys.reduce((memo, key, index) => ({
...memo,
[key.name] = values[index],
});

Reply:
it's code from react-router v4, i don't think it's usefull to replace it. As it's now, we have this file, but when migration on v4 would be done, this file would be replace by the one from the dependency

Reply:
ok